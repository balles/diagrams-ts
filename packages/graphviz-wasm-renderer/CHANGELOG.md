# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.10](https://github.com/balles/diagrams-ts/compare/v0.3.9...v0.3.10) (2021-05-04)

**Note:** Version bump only for package @diagrams-ts/graphviz-wasm-renderer





## [0.3.7](https://github.com/balles/diagrams-ts/compare/v0.3.6...v0.3.7) (2021-05-02)

**Note:** Version bump only for package @diagrams-ts/graphviz-wasm-renderer





## [0.3.6](https://github.com/balles/diagrams-ts/compare/v0.3.5...v0.3.6) (2021-05-02)

**Note:** Version bump only for package @diagrams-ts/graphviz-wasm-renderer





## [0.3.5](https://github.com/balles/diagrams-ts/compare/v0.3.4...v0.3.5) (2021-04-29)

**Note:** Version bump only for package @diagrams-ts/graphviz-wasm-renderer





## [0.3.4](https://github.com/balles/diagrams-ts/compare/v0.3.3...v0.3.4) (2021-04-28)

**Note:** Version bump only for package @diagrams-ts/graphviz-wasm-renderer





# [0.3.0](https://github.com/balles/diagrams-ts/compare/v0.2.1...v0.3.0) (2021-04-21)


### Features

* **core:** move core function in separate package ([#33](https://github.com/balles/diagrams-ts/issues/33)) ([e02019a](https://github.com/balles/diagrams-ts/commit/e02019aa2e00d95582313e50fdd7e49138a7e6f3))





# [0.2.0](https://github.com/balles/diagrams-ts/compare/v0.1.0...v0.2.0) (2021-04-20)

**Note:** Version bump only for package @diagrams-ts/graphviz-wasm-renderer





# [0.1.0](https://github.com/balles/diagrams-ts/compare/v0.0.5...v0.1.0) (2021-04-18)


### Features

* **renderer:** add wasm renderer ([#30](https://github.com/balles/diagrams-ts/issues/30)) ([4eb9472](https://github.com/balles/diagrams-ts/commit/4eb9472b5c6b206ee6ddc455f0748fc4fb375f82))
