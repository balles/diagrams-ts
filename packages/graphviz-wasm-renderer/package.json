{
  "name": "@diagrams-ts/graphviz-wasm-renderer",
  "version": "0.5.0",
  "description": " render dot-language string to svg with a graphviz compiled to webassembly",
  "main": "dist/wasm-renderer.js",
  "types": "dist/wasm-renderer.d.ts",
  "author": "balles",
  "license": "MIT",
  "private": false,
  "repository": {
    "type": "git",
    "url": "https://github.com/balles/diagrams-ts.git",
    "directory": "packages/graphviz-wasm-renderer"
  },
  "scripts": {
    "build": "yarn tsc",
    "test": "yarn jest ./src",
    "lint": "yarn prettier . --check && yarn eslint . --ext .ts",
    "format": "yarn prettier . --write"
  },
  "devDependencies": {
    "@types/node": "15.0.1",
    "@typescript-eslint/eslint-plugin": "4.22.0",
    "@typescript-eslint/parser": "4.22.0",
    "eslint": "7.25.0",
    "jest": "26.6.3",
    "lerna": "^3.22.0",
    "prettier": "2.2.1",
    "ts-jest": "26.5.5",
    "typescript": "4.3.2"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@hpcc-js/wasm": "1.5.1"
  }
}
