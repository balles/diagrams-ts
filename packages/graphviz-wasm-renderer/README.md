# GraphViz-WASM-Renderer

Use a graphviz version compiled to web assembly, to render a string containing valid `dot` syntax to an image output. Can only render to `svg` and other text based image output (not `png`,`webp`,...)

## Installation and Requirements

Install this package by running:

```sh
npm install @diagrams-ts/graphviz-wasm-renderer

# or when using yarn

yarn add @diagrams-ts/graphviz-wasm-renderer
```

If you are packaging with Webpack or a similar bundler make sure that the WASM files are included.

## Usage

```ts
import { WasmRenderer } from "@diagrams-ts/graphviz-wasm-renderer";

const render = WasmRenderer({ outputFile: "./example.svg", format: "svg" });

(async () => {
  try {
    await render(
      `digraph G {
        a1 -> b2;
        a1 -> b3;
    }`
    );
  } catch (error) {
    console.log(error);
  }
})();
```

If you want to use it outside of `NodeJS` you can use it to render to a string instead:

```ts
import { WasmToStringRenderer } from "@diagrams-ts/graphviz-wasm-renderer";

const render = WasmToStringRenderer({ format: "svg" });

(async () => {
  try {
    const mySVG = await render(
      `digraph G {
        a1 -> b2;
        a1 -> b3;
    }`
    );
    console.log(mySVG); //Outputs an string with a valid SVG you could embed in your HTML
  } catch (error) {
    console.log(error);
  }
})();
```
