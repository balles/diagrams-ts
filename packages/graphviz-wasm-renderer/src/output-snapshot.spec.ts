// Simple test render, in order to catch unexpected api changes in upstream package
import { WasmToStringRenderer } from "@diagrams-ts/graphviz-wasm-renderer";

it("should render to output file", async () => {
  const dotInput = `digraph G {
    a [ image="https://github.com/balles/diagrams-ts/raw/main/generated-assets/logo-small.png" ]
    a1 -> b2;
    a1 -> b3;
}`;
  expect(
    await WasmToStringRenderer({ format: "svg" })(dotInput)
  ).toMatchSnapshot();
});
