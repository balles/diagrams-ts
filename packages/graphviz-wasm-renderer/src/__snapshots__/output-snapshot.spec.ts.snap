// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render to output file 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\"
 \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<!-- Generated by graphviz version 2.47.0 (20210316.1810)
 -->
<!-- Title: G Pages: 1 -->
<svg width=\\"356pt\\" height=\\"320pt\\"
 viewBox=\\"0.00 0.00 356.21 320.42\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
<g id=\\"graph0\\" class=\\"graph\\" transform=\\"scale(1 1) rotate(0) translate(4 316.42)\\">
<title>G</title>
<polygon fill=\\"white\\" stroke=\\"transparent\\" points=\\"-4,4 -4,-316.42 352.21,-316.42 352.21,4 -4,4\\"/>
<!-- a -->
<g id=\\"node1\\" class=\\"node\\">
<title>a</title>
<ellipse fill=\\"none\\" stroke=\\"black\\" cx=\\"120.21\\" cy=\\"-192.21\\" rx=\\"120.42\\" ry=\\"120.42\\"/>
<image xlink:href=\\"https://github.com/balles/diagrams-ts/raw/main/generated-assets/logo-small.png\\" width=\\"225px\\" height=\\"225px\\" preserveAspectRatio=\\"xMinYMin meet\\" x=\\"7.70815\\" y=\\"-304.708\\"/>
<text text-anchor=\\"middle\\" x=\\"120.21\\" y=\\"-188.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">a</text>
</g>
<!-- a1 -->
<g id=\\"node2\\" class=\\"node\\">
<title>a1</title>
<ellipse fill=\\"none\\" stroke=\\"black\\" cx=\\"285.21\\" cy=\\"-192.21\\" rx=\\"27\\" ry=\\"18\\"/>
<text text-anchor=\\"middle\\" x=\\"285.21\\" y=\\"-188.01\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">a1</text>
</g>
<!-- b2 -->
<g id=\\"node3\\" class=\\"node\\">
<title>b2</title>
<ellipse fill=\\"none\\" stroke=\\"black\\" cx=\\"249.21\\" cy=\\"-18\\" rx=\\"27\\" ry=\\"18\\"/>
<text text-anchor=\\"middle\\" x=\\"249.21\\" y=\\"-13.8\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">b2</text>
</g>
<!-- a1&#45;&gt;b2 -->
<g id=\\"edge1\\" class=\\"edge\\">
<title>a1&#45;&gt;b2</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M281.62,-174.07C275.37,-144.15 262.42,-82.19 254.92,-46.31\\"/>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"258.25,-45.14 252.78,-36.07 251.4,-46.58 258.25,-45.14\\"/>
</g>
<!-- b3 -->
<g id=\\"node4\\" class=\\"node\\">
<title>b3</title>
<ellipse fill=\\"none\\" stroke=\\"black\\" cx=\\"321.21\\" cy=\\"-18\\" rx=\\"27\\" ry=\\"18\\"/>
<text text-anchor=\\"middle\\" x=\\"321.21\\" y=\\"-13.8\\" font-family=\\"Times,serif\\" font-size=\\"14.00\\">b3</text>
</g>
<!-- a1&#45;&gt;b3 -->
<g id=\\"edge2\\" class=\\"edge\\">
<title>a1&#45;&gt;b3</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M288.79,-174.07C295.05,-144.15 308,-82.19 315.5,-46.31\\"/>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"319.02,-46.58 317.64,-36.07 312.17,-45.14 319.02,-46.58\\"/>
</g>
</g>
</svg>
"
`;
